pkg/engine/loader.go

```diff
@@ -74,3 +86,75 @@ func (e *Engine) LoadConstraintsFromFile(path string) (map[constraints.Constrain
 	}
 	return constraints.ParseConstraintsFromFile(bytesArr)
 }
+
+func correctPointers(source core.BaseConstruct, resourceMap map[core.ResourceId]core.BaseConstruct) error {
```

Comment:
Can you add a little documentation to this for what it does and why it's needed?